pkg/controller/alertmanagerconfig/components/alertmanagerconfig.go

```
@@ -0,0 +1,143 @@
+	alertconfig "github.com/prometheus/alertmanager/config"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type alertManageConfigComponent struct {
+}
+
+func NewAlertManagerConfig() *alertManageConfigComponent {
+	return &alertManageConfigComponent{}
+}
+
+func (_ *alertManageConfigComponent) WatchTypes() []runtime.Object {
+	return []runtime.Object{}
+}
+
+func (_ *alertManageConfigComponent) IsReconcilable(_ *components.ComponentContext) bool {
+	return true
+}
+
+func (comp *alertManageConfigComponent) Reconcile(ctx *components.ComponentContext) (components.Result, error) {
+	// get default alertmanger config
+	instance := ctx.Top.(*monitoringv1beta1.AlertManagerConfig)
+	defaultConfigSecret := &corev1.Secret{}
+	err := ctx.Get(ctx.Context, types.NamespacedName{
+		Name:      fmt.Sprintf("%s-default", instance.Spec.AlertManagerName),
+		Namespace: instance.Spec.AlertManagerNamespace}, defaultConfigSecret)
+	if err != nil {
+		return components.Result{}, errors.Wrapf(err, "Failed to get default AlertManagerConfig")
+	}
+	// verify default config
+	defaultConfig, err := alertconfig.Load(string(defaultConfigSecret.Data["alertmanager.yaml"][:]))
+	if err != nil {
+		return components.Result{}, errors.Wrapf(err, "Looks like default config in bad format.")
+	}
+	// Get all AlertManagerConfig's
+	alertList := &monitoringv1beta1.AlertManagerConfigList{}
+	err = ctx.List(ctx.Context, &client.ListOptions{}, alertList)
+	if err != nil {
+		return components.Result{}, errors.Wrapf(err, "failed to list alermanagerconfig.")
+	}
+	// Merge  all AlertManagerConfig's
+	for _, config := range alertList.Items {
+		var route, receiver, inhibitRule []byte
+		routetype := &alertconfig.Route{}
+		receivertype := &alertconfig.Receiver{}
+		inhibitRuletype := &alertconfig.InhibitRule{}
+		if config.Spec.AlertManagerName == instance.Spec.AlertManagerName {
+			route, _ = base64.StdEncoding.DecodeString(config.Spec.Data["routes"])
```

Shouldn't ignore errors here either.